pkg/api/controllers/gitprovider/branches.go

```diff
@@ -43,9 +50,33 @@ func GetRepoBranches(ctx *gin.Context) {
 		return
 	}
 
+	page := 1
+	perPage := 100
+
+	if pageQuery != "" {
```

Comment:
since this is reused a lot, please make function that extract this.

E.g. `listOptions := getListOptions(ctx)`

Reply:
fixed